src/runner.rs

```diff
@@ -26,18 +46,42 @@ pub async fn run_test_command(
         Ok(result) => result,
         Err(e) => {
             log::error!("Error waiting for execution: {}", e);
-            return Ok(RunResult {
-                exit_code: EXIT_FAILURE,
-                failures: Vec::new(),
-            });
+            return Ok(EXIT_FAILURE);
         }
     };
-    let mut failures = Vec::<Test>::new();
-    if !result.success() {
-        let mut file_counter = FileSetCounter::default();
-        let file_sets = match output_paths
+    Ok(result.code().unwrap_or(EXIT_FAILURE))
+}
+
+pub fn get_files(
+    repo: &BundleRepo,
+    junit_paths: Vec<String>,
+    team: Option<String>,
+    codeowners_path: Option<String>,
+) -> anyhow::Result<(Vec<FileSet>, FileSetCounter)> {
+    let mut file_counter = FileSetCounter::default();
+    let mut file_sets = junit_paths
+        .iter()
+        .map(|path| {
+            FileSet::scan_from_glob(
```

Comment:
Outside the scope of this PR, this file counting logic and fallback to checking of nested paths should be a part of the `FileSet` struct. It scan have its own state to do this rather than it living in this function.